FIRO_TSEnsembles/src/main/java/hec/stats/MultiStatComputable.java

```diff
@@ -1,20 +1,51 @@
 package hec.stats;
 
-public class MultiStatComputable implements MultiComputable{
+public class MultiStatComputable implements MultiComputable, Computable {
     Statistics[] statSelection;
 
     /**
      * The MultiComputable interface is beneficial for creating multiple time series representations.
      * This method iterates across all traces for each timestep to produce multiple values for each
      * timestep. A good example would be the max and min for all timesteps which would represent the
-     * bounds of the ensemble.
+     * bounds of the ensemble.  Implements Computable interface to allow class to be used by twoStepComputable
+     * which requires two Computable objects.
      * @param statSelection is expected to be a String
      */
 
     public MultiStatComputable(Statistics[] statSelection) {
         this.statSelection = statSelection;
     }
 
+    @Override
+    public float compute(float[] values) {
+        float results = 0;
+        InlineStats is = new InlineStats();
+
+        for(float f : values){
+            is.AddObservation(f);
+        }
+        switch (statSelection[0]){
+            case MIN:
+                results = is.getMin();
+                break;
+            case MEAN:
+                results = is.getMean();
+                break;
+            case MAX:
+                results = is.getMax();
+                break;
+            case VARIANCE:
+                results = is.getSampleVariance();
+                break;
+            case STANDARDDEVIATION:
+                results = is.getSampleStandardDeviation();
+                break;
+            default:
+                throw new ArithmeticException("stat type not  yet supported in MultiStatComputable.");
+        }
+        return results;
+    }
+
     @Override
     public float[] MultiCompute(float[] values) {
```

Comment:
should be camel case method name since this is Java.
what is different that the compute() method above?    Since the class name includes 'Multi' the method should be compute().

Reply:
implemented the Computable interface so I need to override the compute method.  The compute method and multiCompute method do the same thing.  I added this interface so we could allow this class to be used for the twoStepComputable class, which takes in two Computable objects (cannot take a MultiComputable object).  Not sure if there was a better way to go about it.

Reply:
@dvdhoz  what @ktarbet  is trying to say is write the method once - probably as a private method. and call it from both places. That way you do not implement the same code twice, you just call it twice. Alternatively, as he specifically says, call compute from MultiCompute, which would also be legitimate if the signature allows, but it is up to you.

Reply:
added helper function so switch statement is called once